plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/refactoring/N4JSRenameStrategy.java

```diff
@@ -0,0 +1,170 @@
+				ITextRegion nameRegion = locationInFileProvider.getFullTextRegion(constituentMember, att, 0);
+
+				EObject nameElement = ((SyntaxRelatedTElement) constituentMember).getAstElement();
+
+				// PropertyNameOwner has a LiteralOrComputedPropertyName child node that contains the name to be
+				// renamed
+				if (nameElement instanceof PropertyNameOwner) {
+					nameElement = ((PropertyNameOwner) nameElement).getDeclaredName();
+					EAttribute nAttribute = SimpleAttributeResolver.newResolver(String.class, "literalName")
+							.getAttribute(nameElement);
+					nameRegion = locationInFileProvider.getFullTextRegion(nameElement, nAttribute, 0);
+				}
+
+				ReplaceEdit replaceEdit = new ReplaceEdit(nameRegion.getOffset(), nameRegion.getLength(), text);
+				updateAcceptor.accept(EcoreUtil.getURI(constituentMember).trimFragment(),
+						replaceEdit);
+			}
+
+		} else {
+			updateAcceptor.accept(getTargetElementOriginalURI().trimFragment(), getDeclarationTextEdit(newName));
+		}
+	}
+
+	@Override
+	public void applyDeclarationChange(String newName, ResourceSet resourceSet) {
+		if ((targetElement instanceof TMember) && ((TMember) targetElement).isComposed()) {
+			targetElementNewURIs.clear();
+			List<TMember> constituentMembers = ((TMember) targetElement).getConstituentMembers();
+			for (TMember constituentMember : constituentMembers) {
+				URI constituentMemberURI = EcoreUtil.getURI(constituentMember);
+				EObject renamedElement = setName(constituentMemberURI, newName, resourceSet);
+				targetElementNewURIs.add(EcoreUtil.getURI(renamedElement));
+			}
+		}
+		super.applyDeclarationChange(getNameAsValue(newName), resourceSet);
+	}
+
+	@Override
+	public void revertDeclarationChange(ResourceSet resourceSet) {
+		if ((targetElement instanceof TMember) && ((TMember) targetElement).isComposed()) {
+			for (URI targetElementNewURI : targetElementNewURIs) {
+				setName(targetElementNewURI, getOriginalName(), resourceSet);
+			}
+		}
+		super.revertDeclarationChange(resourceSet);
+	}
+
+	/** Extract the text region of the AST node that contains the name to be renamed */
+	@Override
+	protected ITextRegion getOriginalNameRegion(final EObject targetElement, EAttribute nameAttribute) {
+		if (targetElement instanceof SyntaxRelatedTElement) {
+			EObject nameElement = ((SyntaxRelatedTElement) targetElement).getAstElement();
+
+			// PropertyNameOwner has a LiteralOrComputedPropertyName child node that contains the name to be renamed
+			if (nameElement instanceof PropertyNameOwner) {
+				nameElement = ((PropertyNameOwner) nameElement).getDeclaredName();
+				EAttribute nAttribute = SimpleAttributeResolver.newResolver(String.class, "literalName")
+						.getAttribute(nameElement);
+				return this.getOriginalNameRegion(nameElement, nAttribute);
+			}
+		}
+		return super.getOriginalNameRegion(targetElement, nameAttribute);
+	}
+
+	@Override
+	public RefactoringStatus validateNewName(String newName) {
+		// TODO: Handle n4jsx extension
```

Comment:
seems important

Reply:
Removed, n4jsx was merged into n4js grammar by you :D